network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/NetworkImpl.java

```diff
@@ -95,12 +98,12 @@ public Bus getBus(String id) {
 
         @Override
         public Bus getBus(String id) {
-            throw new UnsupportedOperationException("TODO");
+            return getBusStream().filter(b -> b.getId().equals(id)).findFirst().orElse(null);
         }
 
         @Override
-        public Collection<Component> getConnectedComponents() {
-            throw new UnsupportedOperationException("TODO");
+        public Collection<Component> getConnectedComponents() { // FIXME : need a reference bus by component
+            return getBusStream().map(Bus::getConnectedComponent).collect(Collectors.toList());
```

Comment:
This is not correct, as you are going to have here as many ConnectedComponent object as Bus but the aim was to get a list of unique ConnectedComponent